// athene-tf-test/main.js
(function(exports) {
  var parentElement = document.getElementById("div1");

  var test = {};

  var string = '';
  (function(obj) {

    obj.divId = 0

    obj.with = function(value2) {
      document.getElementById(obj.value).value= value2
    },

    obj.isTrue = function() {
      result(obj.value)
    },

    obj.isEqualTo = function(value2) {
      if (obj.value === value2){
        showResult('');
        result(obj.value === value2);
      } else {
        showResult(`expected "${value2}" to equal "${obj.value}"`);
        result(obj.value === value2);
      }
    },

    obj.toContain = function(value2) {
        showResult('');
        result(obj.value.includes(value2));
    },

    obj.suiteDescription = function(desc) {
      obj.divId += 1
      var describeTitle = document.getElementById("div1")
      describeTitle.innerHTML += "<div id=" + test.divId +"><h4>" + desc + "</h4></div>"

    }

    function showResult(stringv) {
      return string = stringv;
    }

    function result(assertion) {
      var result = (assertion) ? "Pass" : "Fail";
      printTestResult(result);
    }

    function printTestResult(result) {
      var error = string
      var tickOrX = (result === "Pass") ?  "&#10004 " : "&#10008 ";
      var itText = document.getElementById(test.divId)
      itText.innerHTML += "<li class=" + result +"> " + tickOrX + obj.description + "</li>" +
                          "<span class=" + result +"> " + error + "</span>";


    }
  })(test);

  var expect = function(value) {
    test.value = value;
    return test;
  }

  var it = function(testDescription, method) {
    test.description = testDescription;
    method();
  }

  var describe = function(suiteDescription, method) {
    test.suiteDescription(suiteDescription);
    method();
  }

  var fillIn = function(value) {
    test.value = value;
    return test;
  }

  var clickButton = function(button) {
    document.getElementById(button).click();
  }

  exports.test = test;
  exports.expect = expect;
  exports.it = it;
  exports.describe = describe;
  exports.fillIn = fillIn;
  exports.clickButton = clickButton;

})(this);
